doc/man3/ASYNC_WAIT_CTX_new.pod

```diff
@@ -103,14 +114,37 @@ code that the job should be resumed by making the wait file descriptor
 "readable". Once resumed the engine should clear the wake signal on the wait
 file descriptor.
 
+Besides file descriptor, an optional additional method of notification is
+provided via callback owned by user code. The callback and data pointers would
+be stored within the ASYNC_WAIT_CTX along with an additional status field that
+can be used for the notification of retries from an engine. This additional
+method can be used when user thinks the file descriptor is too costly in terms
+of CPU cycles or in some context where file descriptor is not appropriate.
```

Comment:
That's good point. The design has already considered this situation. Each time after user code sends out the request to engine,  it will check the status by calling ASYNC_WAIT_CTX_get_status(). See below for the detailed return value. The "ASYNC_STATUS_UNSUPPORTED" is to indicated that engine does not have any idea of "callback" as it is still the initial value of 0. So a smart user code know this is "file descriptor" only engine, and then use file descriptor for communication.

Reply:
I also added some info at the "status" part to make people understand it.